app/routes.py

```diff
@@ -1 +1,205 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint,jsonify, request, make_response, abort
+import requests
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+import datetime
+import os
+
+
+task_bp = Blueprint("tasks", __name__,url_prefix="/tasks")
+
+#post a task
+@task_bp.route("", methods=["POST"])
+def add_task():
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body:
+        return {"details": "Bad Request"}, 400
+    
+    new_task = Task.from_dict(request_body)
+    
+
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task" : new_task.to_dict()}, 201
+
+@task_bp.route("", methods=["GET"])
+def get_all_tasks():
+    response = []
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc":
+        all_tasks = Task.query.order_by(Task.title.asc())
+    elif sort_query == "desc":
+        all_tasks = Task.query.order_by(Task.title.desc())
+    else:
+        all_tasks = Task.query.all()
+
+    for task in all_tasks:
+        response.append(task.to_dict())
+
+    return jsonify(response), 200
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    task = validate_id(Task,task_id)
+
+    return {"task" : task.to_dict()}, 200
+
+@task_bp.route("<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_id(Task, task_id)
+
+    request_data = request.get_json()
+
+    task.title = request_data["title"]
+    task.description = request_data["description"]
```

Comment:
I'd recommend wrapping these lines in a try-except block so that you can return an error message if the user is missing some of the required data